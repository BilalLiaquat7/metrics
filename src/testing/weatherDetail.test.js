import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import WeatherDetail from '../components/weatherDetail';

const mockStore = configureStore([]);

describe('WeatherDetail Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      weather: {
        weather: [
          {
            id: '1',
            name: 'Pakistan',
            main: { temp: 25 },
            wind: { speed: 10, deg: 90 },
            weather: [{ description: 'Sunny' }],
          },
        ],
      },
    });
  });

  test('renders component with valid country data', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <WeatherDetail cityName="Pakistan" />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders component with invalid country data', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <WeatherDetail cityName="InvalidCountry" />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
