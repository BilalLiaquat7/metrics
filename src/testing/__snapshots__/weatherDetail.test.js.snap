// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetail Component renders component with invalid country data 1`] = `
<div>
  <div
    class="countryDetailMain"
  />
</div>
`;

exports[`WeatherDetail Component renders component with valid country data 1`] = `
<div>
  <div
    class="countryDetailMain"
  />
</div>
`;
